src/clarity-angular/datagrid/selection-type.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+export enum SelectionType {
```

Comment:
This feels out of place here. I think you should just put this enum in `providers/selection.ts`, that's the most logical place for it.

I know it's good practice is to have one class per file, but constants, enums or small interfaces directly linked to a class can be in the same file without it becoming pollution. It actually makes things clearer.